railties/lib/rails/commands/server.rb

```diff
@@ -94,7 +94,8 @@ def default_options
         environment:        (ENV['RAILS_ENV'] || ENV['RACK_ENV'] || "development").dup,
         daemonize:          false,
         caching:            nil,
-        pid:                Options::DEFAULT_PID_PATH
+        pid:                Options::DEFAULT_PID_PATH,
+        restart_cmd:        'bin/rails s'
```

Comment:
@rubys No. It will start on port 3000 instead of 4567.

```
$ rails server -b 0.0.0.0 -p 4567
=> Booting Puma
=> Rails 5.0.0.beta3 application starting in development on http://0.0.0.0:4567
=> Run `rails server -h` for more startup options
Puma starting in single mode...
* Version 3.2.0 (ruby 2.3.0-p0), codename: Spring Is A Heliocentric Viewpoint
* Min threads: 5, max threads: 5
* Environment: development
* Listening on tcp://0.0.0.0:4567
Use Ctrl-C to stop
* Restarting...
=> Booting Puma
=> Rails 5.0.0.beta3 application starting in development on http://localhost:3000
=> Run `rails server -h` for more startup options
Puma starting in single mode...
* Version 3.2.0 (ruby 2.3.0-p0), codename: Spring Is A Heliocentric Viewpoint
* Min threads: 5, max threads: 5
* Environment: development
* Listening on tcp://localhost:3000
* Closing unused inherited connection: tcp://0.0.0.0:4567
Use Ctrl-C to stop
```

I hadn't thought of this scenario. I will check how to make it work.